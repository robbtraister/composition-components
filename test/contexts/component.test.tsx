'use strict'

/* global describe, expect, test */

import PropTypes from 'prop-types'
import React from 'react'
import { render } from '@testing-library/react'

import App from '../../src/components/app'
import {
  ComponentContext,
  useComponentContext
} from '../../src/contexts/component'

function Display({ context }) {
  return <div>{context.id}</div>
}

Display.propTypes = {
  context: PropTypes.shape({
    id: PropTypes.string
  })
}

const ContextComponent = () => {
  return <ComponentContext component={Display} />
}

const ContextHook = () => {
  const context = useComponentContext()
  return <Display context={context} />
}

const ContextRender = () => {
  return <ComponentContext render={Display} />
}

function getComponent(type) {
  switch (type) {
    case 'context-hook':
      return ContextHook
    case 'context-component':
      return ContextComponent
    case 'context-render':
      return ContextRender
  }
}

const node = {
  id: 'abc'
}

function testContextComponent(componentType) {
  return render(
    <App getComponent={getComponent} node={{ ...node, type: componentType }} />
  )
}

describe('Context', () => {
  test('Context Component', async () => {
    const { asFragment } = testContextComponent('context-component')
    expect(asFragment()).toMatchSnapshot()
  })

  test('Context Hook', async () => {
    const { asFragment } = testContextComponent('context-hook')
    expect(asFragment()).toMatchSnapshot()
  })

  test('Context Render', async () => {
    const { asFragment } = testContextComponent('context-render')
    expect(asFragment()).toMatchSnapshot()
  })
})
