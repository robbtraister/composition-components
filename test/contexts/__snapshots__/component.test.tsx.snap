// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context Context Component 1`] = `
<DocumentFragment>
  <div>
    abc
  </div>
</DocumentFragment>
`;

exports[`Context Context Hook 1`] = `
<DocumentFragment>
  <div>
    abc
  </div>
</DocumentFragment>
`;

exports[`Context Context Render 1`] = `
<DocumentFragment>
  <div>
    abc
  </div>
</DocumentFragment>
`;
